opal/corelib/string/unpack.rb

```diff
@@ -686,6 +686,15 @@ class String
       'p': null
     }
 
+    var optimized = {
+      'C*': identityFunction,
```

Comment:
Hmm, didn't expect it to be that composable 😄 
Now I start thinking that probably it makes sense to use references to handlers here (because what you've got is a true code duplication).
```
{
  'C*': handlers['C'],
  'c*': handlers['c'],
  ...
}
```

But the way, how fast is it? It can be optimized even more with inline functions:
``` js
'A*': function(buffer) {
  var string = String.fromCharCode.apply(null, buffer);
  string = string.replace(/[\u0000 ]+$/, '');
  return [string];
},
'a*': function(buffer) {
  return String.fromCharCode.apply(null, buffer);
} 
```

Reply:
Moreover, I guess case for `CHAR*` can be generalized for **all** chars. I'm checking it locally

Reply:
No, some of them wrap result into array (you just pushed a function `toArray`):
``` fails "String#unpack with format 'B' decodes all the bits when passed the '*' modifier" # Expected [""] to be computed by "".unpack from "B*" (computed "" instead)
fails "String#unpack with format 'H' decodes all the nibbles when passed the '*' modifier" # Expected [""] to be computed by "".unpack from "H*" (computed "" instead)
fails "String#unpack with format 'M' ignores the count or '*' modifier and decodes the entire string" # Expected ["abc"] to be computed by "a=\nb=\nc=\n".unpack from "M*" (computed "abc" instead)
fails "String#unpack with format 'U' raises ArgumentError on a malformed byte sequence and doesn't continue when used with the * modifier" # Expected ArgumentError but no exception was raised ([227] was returned)
fails "String#unpack with format 'Z' decodes past whitespace bytes when passed the '*' modifier" # Expected ["a b c"] to be computed by "a b c".unpack from "Z*" (computed "a b c" instead)
fails "String#unpack with format 'b' decodes all the bits when passed the '*' modifier" # Expected [""] to be computed by "".unpack from "b*" (computed "" instead)
fails "String#unpack with format 'h' decodes all the nibbles when passed the '*' modifier" # Expected [""] to be computed by "".unpack from "h*" (computed "" instead)
fails "String#unpack with format 'm' ignores the count or '*' modifier and decodes the entire string" # Expected ["abcDEF"] to be computed by "YWJj\nREVG\n".unpack from "m*" (computed "abcDEF" instead)
fails "String#unpack with format 'u' ignores the count or '*' modifier and decodes the entire string" # Expected ["abcDEF"] to be computed by "#86)C\n#1$5&\n".unpack from "u*" (computed "abcDEF" instead)
```

Reply:
> Now I start thinking that probably it makes sense to use references to handlers here (because what you've got is a true code duplication).

True 😐 

> But the way, how fast is it? It can be optimized even more with inline functions:

Hmmm, interesting... 
> Moreover, I guess case for CHAR* can be generalized for all chars. I'm checking it locally

I think that's what I did, if the buffer remains intact then we can use directly the handler function.

Reply:
I was wrong, there's also an autocompletion rule 😄 I love this method, guys from MRI completely ignore SRP.

> True 😐

so, how about
```
{
  'C*': handlers['C'],
  'c*': handlers['c'],
  ...
  'a*': wrapIntoArray(handlers['a'])
}
```

Reply:
> so, how about...

Done!